parcels/field.py

```diff
@@ -116,13 +116,18 @@ def __init__(self, name, data, lon=None, lat=None, depth=None, time=None, grid=N
             self.allow_time_extrapolation = allow_time_extrapolation
 
         self.time_periodic = time_periodic
-        if self.time_periodic and self.allow_time_extrapolation:
+        if self.time_periodic is not False and self.allow_time_extrapolation:
             logger.warning_once("allow_time_extrapolation and time_periodic cannot be used together.\n \
                                  allow_time_extrapolation is set to False")
             self.allow_time_extrapolation = False
-        if self.time_periodic:
-            logger.warning_once("When using time_periodic=True, it is necessary that the first and last time steps\n \
-                                 of the series are the same, with time[-1] = time[0] + T")
+        if self.time_periodic is True:
+            raise ValueError("Unsupported time_periodic=True. time_periodic must now be either False or the period value")
+        if self.time_periodic is not False and not np.isclose(self.grid.time[-1] - self.grid.time[0], self.time_periodic):
+            if self.grid.time[-1] - self.grid.time[0] > self.time_periodic:
+                raise ValueError("Time series provided is longer than the time_periodic parameter")
+            self.grid._add_last_periodic_data_timestep = True
+            self.grid.time = np.append(self.grid.time, self.grid.time[0] + self.time_periodic)
+            self.grid.time_full = self.grid.time
```

Comment:
This shows how confusing `grid.time` and `grid.time_full` are. We really should at some point fix this, see also #651 (although not in this PR)

Reply:
Yes indeed!